extension/intents/nicknames/routineExecutor.js

```diff
@@ -0,0 +1,48 @@
+/* globals log */
+
+import * as intentRunner from "../../background/intentRunner.js";
+
+export class RoutineExecutor {
+  constructor(routineName, subcommands, index = 0) {
+    this.routineName = routineName;
+    this.subcommands = subcommands;
+    this.index = index;
+  }
+
+  async runSubcommand(subcommand) {
+    let hadError = false;
+    subcommand.onError = () => {
+      hadError = true;
```

Comment:
This should capture the message too, like `subcommand.onError = (message) => {hadError=true; errorMessage=message}`